### PR TITLE
Add 7.1-preview service features for Key Vault 4.1-preview.1

### PR DESCRIPTION
**Pull request by heaths**

### CODE REVIEW

**Comment by heaths** [comment]
Added "Do Not Merge" label until we get the servicing version released and this rebased (some CHANGELOGs will need to be updated manually).

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/api/Azure.Security.KeyVault.Certificates.netstandard2.0.cs [hunk 1]
```diff
@@ -188,6 +188,7 @@ public static partial class CertificateModelFactory
...
         public static Azure.Security.KeyVault.Certificates.CertificateProperties CertificateProperties(System.Uri id = null, string name = null, System.Uri vaultUri = null, string version = null, byte[] x509thumbprint = null, System.DateTimeOffset? notBefore = default(System.DateTimeOffset?), System.DateTimeOffset? expiresOn = default(System.DateTimeOffset?), System.DateTimeOffset? createdOn = default(System.DateTimeOffset?), System.DateTimeOffset? updatedOn = default(System.DateTimeOffset?), string recoveryLevel = null) { throw null; }
+        public static Azure.Security.KeyVault.Certificates.CertificateProperties CertificateProperties(System.Uri id = null, string name = null, System.Uri vaultUri = null, string version = null, byte[] x509thumbprint = null, System.DateTimeOffset? notBefore = default(System.DateTimeOffset?), System.DateTimeOffset? expiresOn = default(System.DateTimeOffset?), System.DateTimeOffset? createdOn = default(System.DateTimeOffset?), System.DateTimeOffset? updatedOn = default(System.DateTimeOffset?), string recoveryLevel = null, int? recoverableDays = default(int?)) { throw null; }
```
Comment (tg-msft) [inline sdk/keyvault/Azure.Security.KeyVault.Certificates/api/Azure.Security.KeyVault.Certificates.netstandard2.0.cs]: Nit - should we hide the old one with `[EditorBrowsable(Never)]` and also remove all the default values?

Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Certificates/api/Azure.Security.KeyVault.Certificates.netstandard2.0.cs]: Hiding the old one wouldn't hurt, but based on the example we should use default values when, it seems, those values are optional (in our case, that's most of them for models): https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-mocking

Comment (tg-msft) [reply sdk/keyvault/Azure.Security.KeyVault.Certificates/api/Azure.Security.KeyVault.Certificates.netstandard2.0.cs]: I only mean remove the default values from the old API that you'd hide via `[EditorBrowsable(Never)]`.  Any code with optional values would change to the new API as soon as they recompiled.

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateClientOptions.cs [hunk 2]
```diff
@@ -56,13 +56,12 @@ public CertificateClientOptions(ServiceVersion version = ServiceVersion.V7_0)
 
         internal string GetVersionString()
         {
-            var version = Version switch
+            return Version switch
             {
                 ServiceVersion.V7_0 => "7.0",
 
                 _ => throw new ArgumentException(Version.ToString()),
```
Comment (schaabs) [inline sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateClientOptions.cs]: Don't we need to add 7.1-preview support here?

File: sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/CryptographyClientOptions.cs [hunk 3]
```diff
@@ -47,16 +52,21 @@ public enum ServiceVersion
         /// The <see cref="ServiceVersion"/> of the service API used when
         /// making requests.
         /// </param>
+#pragma warning disable AZC0010 // ClientOptions constructors should default ServiceVersion to latest supported service version
         public CryptographyClientOptions(ServiceVersion version = ServiceVersion.V7_0)
+#pragma warning restore AZC0010 // ClientOptions constructors should default ServiceVersion to latest supported service version
```
Comment (schaabs) [inline sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/CryptographyClientOptions.cs]: Should this default to V7_1_Preview?

File: sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClientOptions.cs [hunk 4]
```diff
@@ -48,7 +52,9 @@ public enum ServiceVersion
         /// The <see cref="ServiceVersion"/> of the service API used when
         /// making requests.
         /// </param>
+#pragma warning disable AZC0010 // ClientOptions constructors should default ServiceVersion to latest supported service version
         public KeyClientOptions(ServiceVersion version = ServiceVersion.V7_0)
+#pragma warning restore AZC0010 // ClientOptions constructors should default ServiceVersion to latest supported service version
```
Comment (schaabs) [inline sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClientOptions.cs]: Again shouldn't this default to V7_1_Preview

File: sdk/keyvault/Azure.Security.KeyVault.Keys/CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,11 @@
 # Release History
 
+## 4.1.0-preview.1 (2020-03)
+
+### Added
+
+- Add `RecoverableDays` property to `KeyProperties`.
+
```
Comment (g2vinay) [inline sdk/keyvault/Azure.Security.KeyVault.Keys/CHANGELOG.md]: should add import key op too ?

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/api/Azure.Security.KeyVault.Certificates.netstandard2.0.cs (hunk 1 above)
Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Certificates/api/Azure.Security.KeyVault.Certificates.netstandard2.0.cs]: I opened https://github.com/Azure/azure-sdk/pull/1062 to track changing the requirements with this idea.

**Review by schaabs — APPROVED** [review]

**Pull request merged**